composer/trainer/trainer.py

```diff
@@ -568,6 +569,22 @@ def __init__(
         if num_optimizers != 1:
             raise NotImplementedError(f"Only one optimizer is supported; found {num_optimizers} optimizers")
 
+        # Set initial grad_accum to 1 if using adaptive
+        self.adaptive_grad_accum = isinstance(grad_accum, str) and grad_accum == "auto"
+        if self.adaptive_grad_accum:
```

Comment:
Let's also add a check in here that throws an error if the per-GPU microbatch size is not a power of 2.

Reply:
Why does it have to be a power of 2? I think microbatch size can even be different per GPU (eg say I want a logical batch size of 2001 on 2 GPUs. It seems to work fine right now)